utils/hct/hctdb.py

```diff
@@ -685,12 +692,20 @@ def populate_categories_and_models(self):
             self.name_idx[i].category = "Work Graph intrinsics"
             self.name_idx[i].shader_model = 6, 8
             self.name_idx[i].shader_stages = ("node",)
-        for (
-            i
-        ) in "BarrierByMemoryType,BarrierByMemoryHandle,BarrierByNodeRecordHandle".split(
-            ","
-        ):  # included in Synchronization category
+        for i in "BarrierByMemoryType".split(","):
+            # included in Synchronization category
+            self.name_idx[i].is_barrier = True
+            self.name_idx[i].shader_model = 6, 8
+            self.name_idx[i].shader_model_translated = 6, 0
+        for i in "BarrierByMemoryHandle".split(","):
+            # included in Synchronization category
+            self.name_idx[i].is_barrier = True
+            self.name_idx[i].shader_model = 6, 8
+        for i in "BarrierByNodeRecordHandle".split(","):
+            # included in Synchronization category
+            self.name_idx[i].is_barrier = True
```

Comment:
True for these, the prior one was to capture the flag for all barrier ops, including the pre-existing one.  We could remove it here if you want.  The reason the earlier addition is where it is was due to where `Barrier` was placed in the `Synchronization` category.  That could be moved, but it wouldn't make any difference of course.

Reply:
I also don't like this assignment method, the for loop isn't really being used. Personally I would just say b = "Barrier" and sub in b for the name_idx index, so that the for loops can all be removed. Different variables for the different strings too.